conda/gateways/connection/adapters/ftp.py

```diff
@@ -266,10 +268,13 @@ def build_response(request, data, code, encoding):
     return response
 
 
+@deprecated("24.3", "24.9")
 def parse_multipart_files(request):
     """Given a prepared request, return a file-like object containing the
     original data. This is pretty hacky.
     """
+    import cgi
+
     # Start by grabbing the pdict.
     _, pdict = cgi.parse_header(request.headers["Content-Type"])
```

Comment:
Let's not deprecate part of the ftp adapters that way (especially since we'll have a way to remove it outright when #13054 merges) and instead switch to the email standard library as described in https://peps.python.org/pep-0594/#cgi

For the header parsing:

```suggestion
from email.message import Message

def parse_separated_header(content_type: str):
    # Adapted from https://peps.python.org/pep-0594/#cgi
    message = Message()
    message['Content-Type'] = content_type
    return dict(message.get_params())


def parse_multipart_files(request):
    """Given a prepared request, return a file-like object containing the
    original data. This is pretty hacky.
    """
    # Start by grabbing the pdict.
    pdict = parse_separated_header(request.headers["Content-Type"])
```

The multipart parsing is a little harder I think, I haven't looked into it though